tests/DCPS/AddressLookup/main.cpp

```diff
@@ -0,0 +1,160 @@
+    return;
+  }
+
+  for (addrinfo* curr = res; curr; curr = curr->ai_next) {
+    if (curr->ai_family != AF_INET && curr->ai_family != AF_INET6) {
+      ACE_DEBUG((LM_DEBUG, "hostname_to_ip: Encounter an address that is not AF_INET or AF_INET6\n"));
+      continue;
+    }
+
+    union ip46 {
+        sockaddr_in  in4_;
+#ifdef ACE_HAS_IPV6
+        sockaddr_in6 in6_;
+#endif /* ACE_HAS_IPV6 */
+    } addr;
+    std::memset(&addr, 0, sizeof(addr));
+
+#ifdef ACE_HAS_IPV6
+    ACE_DEBUG((LM_DEBUG, "ip46.in6_ size is %d\n", sizeof(addr.in6_)));
+#endif
+    ACE_DEBUG((LM_DEBUG, "addr size is %d, curr addrinfo size is %d, curr->ai_addrlen is %d, curr->ai_addr size is %d, curr family is %d\n",
+               sizeof(addr), sizeof(*curr), curr->ai_addrlen, sizeof(*(curr->ai_addr)), curr->ai_family));
+    std::memcpy(&addr, curr->ai_addr, curr->ai_addrlen);
+    //std::memcpy(&addr, curr->ai_addr, sizeof addr);
+
+#ifdef ACE_HAS_IPV6
+    if (curr->ai_family == AF_INET6) {
+      addr.in6_.sin6_port = ACE_NTOHS(port_number);
+    } else {
+#endif /* ACE_HAS_IPV6 */
+      addr.in4_.sin_port = ACE_NTOHS(port_number);
+#ifdef ACE_HAS_IPV6
+    }
+#endif /* ACE_HAS_IPV6 */
+
+    ACE_INET_Addr temp;
+    temp.set_addr(&addr, sizeof(addr));
+    temp.set_port_number(port_number, 1 /*encode*/);
+
+    print_addr(temp, "==== IP address:");
+  }
+  ACE_OS::freeaddrinfo(res);
+}
+
+void address_info() {
+  size_t addr_count;
+  ACE_INET_Addr *addr_array = 0;
+  const int result = ACE::get_ip_interfaces(addr_count, addr_array);
+  if (result != 0 || addr_count < 1) {
+    ACE_ERROR((LM_ERROR, "ERROR: print_address_info: ACE::get_ip_interfaces: Unable to probe network interfaces\n"));
+    return;
+  }
+
+  ACE_DEBUG((LM_DEBUG, "address_info: There are %d interfaces\n", addr_count));
+  for (size_t i = 0; i < addr_count; ++i) {
+      char buffer[256] = { '\0' };
+      if (addr_array[i].get_host_addr(buffer, sizeof(buffer)) == 0) {
+          ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to convert address to string\n"));
+      }
+      else {
+          ACE_DEBUG((LM_DEBUG, "DEBUG: address_info: Found IP interface %C\n", buffer));
+      }
+  }
+
+  for (size_t i = 0; i < addr_count; ++i) {
+    ACE_DEBUG((LM_DEBUG, "address_info: Considering interface %d\n", i));
+    //ACE_TCHAR buffer[256];
+    //if (addr_array[i].addr_to_string(buffer, sizeof buffer) != 0) {
+    char buffer[256] = {'\0'};
+    if (addr_array[i].get_host_addr(buffer, sizeof(buffer)) == 0) {
+      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to convert address to string\n"));
+    }
+
+    // Find the hostname of the interface
+    char hostname[MAXHOSTNAMELEN+1] = {'\0'};
+    if (ACE::get_fqdn(addr_array[i], hostname, MAXHOSTNAMELEN+1) == 0) {
+      ACE_DEBUG((LM_DEBUG, "DEBUG: address_info: IP address %C maps to hostname %C\n", buffer, hostname));
+    } else {
+      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to get FQDN\n"));
+    }
```

Comment:
```suggestion
    } else {
      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to get FQDN\n"));
      continue;
    }
```
Skip this interface if it fails to get FQDN.